src/common/helpers/search.js

```diff
@@ -374,8 +384,22 @@ export async function generateIndex(orm) {
 	const entityLists = await Promise.all(behaviorPromise);
 
 	const listIndexes = [];
-	for (const entityList of entityLists) {
+	for (const [index, entityList] of entityLists.entries()) {
 		const listArray = entityList.toJSON();
+		if (index === (entityLists.length - 1)) {
+			for (const workEntity of listArray) {
+				if (workEntity.relationshipSetId) {
+					for (const relationship of workEntity.relationshipSet.relationships) {
+						if (relationship.typeId === 8) {
+							// eslint-disable-next-line no-await-in-loop
+							const source = await getEntityWithAlias(orm, relationship.sourceBbid, 'Author');
```

Comment:
Considering we already load all the Authors, I wonder if we can find the relevant author by bbid from the entityLists instead of loading it again from the DB

Reply:
I doubt iterating over all authors would be better than single db query.